### ISSUE TITLE
Sonos binding: channels not set after a restart

### ISSUE THREAD

**Issue by lolodomo** [eclipse/smarthome#1901]
It looks like sometimes (most of times in fact), after a restart of openHAB; the Sonos binding is not setting the channels anymore.
Maybe the registration with the Sonos fails ?

Tested with beta 3.

**Comment by lolodomo** [comment]
I have the feeling that GENA subscription is simply not working always at startup.
We should have something that checks that we receive something from the events we subscribed to. If not, at least a warning should be logged.

**Comment by lolodomo** [comment]
Just an idea (something to check): maybe the services are not unregistered correctly when OH2 is shutting down.

**Comment by lolodomo** [comment]
I enabled the DEBUG log for org.eclipse.smarthome.io.transport.upnp and I got interesting log entries:
```
10:05:20.209 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Starting UPnP IO service...
10:05:20.774 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Registering device '(RemoteDeviceIdentity) UDN: uuid:RINCON_000E58D01E9001400, Descriptor: http://192.168.0.129:1400/xml/device_description.xml' for participant 'RINCON_000E58D01E9001400'
10:05:20.808 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Registering device '(RemoteDeviceIdentity) UDN: uuid:RINCON_000E588EAE6601400, Descriptor: http://192.168.0.130:1400/xml/device_description.xml' for participant 'RINCON_000E588EAE6601400'
10:05:20.863 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Invoking Action 'GetZoneInfo' of service 'DeviceProperties' for participant 'RINCON_000E58D01E9001400'
10:05:20.926 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Invoking Action 'GetZoneInfo' of service 'DeviceProperties' for participant 'RINCON_000E588EAE6601400'
10:05:31.380 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Current state of service prevents invoking that action. Connection error or no response received.
10:05:31.381 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:RenderingControl' for device 'uuid:RINCON_000E588EAE6601400' failed
10:05:31.382 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:ZoneGroupTopology' for device 'uuid:RINCON_000E58D01E9001400' failed
10:05:31.383 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Invoking Action 'GetZoneAttributes' of service 'DeviceProperties' for participant 'RINCON_000E588EAE6601400'
10:05:31.395 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:AVTransport' for device 'uuid:RINCON_000E58D01E9001400' failed
10:05:31.395 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:GroupManagement' for device 'uuid:RINCON_000E58D01E9001400' failed
10:05:31.397 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:ZoneGroupTopology' for device 'uuid:RINCON_000E588EAE6601400' failed
10:05:31.397 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:AudioIn' for device 'uuid:RINCON_000E588EAE6601400' failed
10:05:31.397 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:AudioIn' for device 'uuid:RINCON_000E58D01E9001400' failed
10:05:31.402 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:AVTransport' for device 'uuid:RINCON_000E588EAE6601400' failed
10:05:31.397 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:RenderingControl' for device 'uuid:RINCON_000E58D01E9001400' failed
10:05:31.405 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:GroupManagement' for device 'uuid:RINCON_000E588EAE6601400' failed
10:05:31.404 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:DeviceProperties' for device 'uuid:RINCON_000E58D01E9001400' failed
10:05:31.407 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:DeviceProperties' for device 'uuid:RINCON_000E588EAE6601400' failed
10:05:31.411 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Current state of service prevents invoking that action. Connection error or no response received.
```
All GENA subscriptions failed.
And there is this error "Current state of service prevents invoking that action. Connection error or no response received".
But some invoked actions are working.

**Comment by lolodomo** [comment]
GENA subscriptions are requested in onUpdate method and so while initialize() is called by the framework. Is it too early for any reason ?

Should we not delay the starting of the polling job of few seconds in order to be sure that the GENA subscriptions are ok when the first actions are invoked ?

**Comment by kgoderis** [comment]
Mmhhhh *normally* these two things should be related as the subscriptions are for data transfer from the UPNP device to openhab, whilst the polling is done from openhab to the UPNP device 
You can invoke an Action at all times ef without subscription if I am not mistaking...

K

**Comment by lolodomo** [comment]
I think that method "failed" in UpnpSubscriptionCallback should trigger something in the UPnP participant to inform it that the subscription failed ? Same thing for method "established" ?

In the method "failed", I think that would better to have a warning log rather than a debug log ?

@kaikreuzer : please enter in the discussion, the problem could be either in the Sonos binding or in the UPnP transport service.

**Comment by lolodomo** [comment]
@kgoderis : yes, I know, you are right.
Why my first action invokation failed ?

PS: of course, I have sometimes a normal startup with no errors and with GENA subscriptions working.

**Comment by lolodomo** [comment]
Maybe the GENA subscriptions should be requested by a (delayed) thread ?

This is done while the thing is even not yet ONLINE.

I would be curious to know if it can help if I delayed the polling job by 1 second and request the GENA subscriptions in another thread delayed by few seconds.

**Comment by lolodomo** [comment]
I tried to put GENA subscriptions in a delayed thread but unfortunately it does not avoid random failing subscriptions !

**Comment by lolodomo** [comment]
I updated the UPnP transport bundle to get the error message:
```
13:01:00.199 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:AVTransport' for device 'uuid:RINCON_000E58D01E9001400' failed: Subscription failed:  No response received.
```
Could it be a too short timeout used by jupnp ?

**Comment by lolodomo** [comment]
@kaikreuzer : what HTTP timeout is used in jupnp ?
Can I update it through eclipse smarthome code or a setting ?

**Comment by kaikreuzer** [comment]
> what HTTP timeout is used in jupnp ?

I don't know, probably one defined in the UPnP specs.

> Can I update it through eclipse smarthome code or even a setting ?

No, I am not aware of such a setting, but I also doubt that it will make any difference. I think it is rather the Sonos speaker that does not respond in certain constellations, so the bug is probably rather in the Sonos firmware than in ESH or jUPnP...

**Comment by lolodomo** [comment]
The GENA subscription I implemented for the Vera Sonos plugin is very reliable. So I don't think the problem is in the Sonos firmware.
I am almost sure that the problem is due to the timeout. In the Vera plugin, I am using a 5s timeout. I think (that is old now and I am not sure) that I discovered that a 2s timeout was not enough.

What do you think about my idea to have a way to notify the bindings that a GENA subscription has succeeded or failed ? It looks like there is no way for a binding to know if it succeeded or failed.

**Comment by kaikreuzer** [comment]
Sounds ok to me, but as I never dealt with Sonos and GENA myself in depth, I would like to leave the further discussion to you, @kgoderis and @R-Edelhoff, since all of you have spent quite some time with that already and hence are the experts.

**Comment by lolodomo** [comment]
Unfortunately, I don't find in jUPnP where are implemented the low level HTTP calls ! I can't control what HTTP timeout is used :(

What I will do as a workaround for the Sonos binding is first implementing #1941 in the UPnP transport service. Then I will include in the Sonos polling job a control of each subscription. If the subscription is not active, the polling job will request a new subscription.

On my RPI, the current success of GENA subscriptions is I think around 33%. By retrying several times, the binding should be able to have valid subscription probably 3 minutes after the binding startup in the worst case.

**Comment by kgoderis** [comment]
Maybe it is wiser to have the jupnp track the subscriptions and maintain them and not have that kind of detail in a/the Sonos binding. Jupnp already has a pool of threads that can be used for this purpose.

**Comment by kgoderis** [comment]
And have callbacks in ioparticipant to notify about changes in the state of a subscription.

**Comment by lolodomo** [comment]
jupnp is certainly dealing with automatic renewing of subscriptions.
In our case, the subscription failed because jupnp probably does not wait enough for a Sonos response. In this case, jupnp can just notify that the subscription failed. So this case cannot be included in jupnp. The thing that could help in jupnp is a way to set the timeout.

**Comment by lolodomo** [comment]
Callbacks in ioparticipant was my initial idea but it will require to update all bindings using the UPnP transport service (at least 3 bindings). What needs to know a binding is only if a subscription is active, that is why I propose to keep the stuff in the transport UPnP service and just provides a method to the bindings that indicates if the subscription is active or not.

**Comment by kgoderis** [comment]
I understand but still I think we should solve it in jupnp.

**Comment by kgoderis** [comment]
Notify callbacks can be optional for a binding to implement....

**Comment by lolodomo** [comment]
Yes, there is absolutely something to fix in jupnp because there is no reason that GENA subscriptionss fail for Sonos. Unfortunately I am not able to fix jupnp, implementation is too much complex for me :(

Yes, any binding could choose to implement or not these callbacks, but at least the override method of the interface should be declared, meaning a minor change in the bindings. If we go to this direction, maybe one unique callback could be enough, notifying if the subscription is active or not.

**Comment by lolodomo** [comment]
Another solution would be to update the transport UPnP service to have the method addSubscription returning if subscription is OK or not. It will require that this method waits for the result notification. But doing that, the binding will immediately knows if its request is OK or not.

**Comment by kgoderis** [comment]
Did you in your installation ever see subscriptions fail after the initial startup sequence?

**Comment by lolodomo** [comment]
Not sure to understand your question.
I am testing the Sonos binding only since few days.
If your question is about subscription renewal, it is something I have not yet checked. I imagine it could fail for the same reason (if my hypothesis is right).
I have no other binding using the transport UPnP service (and so jupnp).

**Comment by lolodomo** [comment]
When I tried to delay the GENA subscriptions, I first thought the problem was solved because I got 2 consecutive success but finally the third startup failed, so I forgave this solution.

**Comment by lolodomo** [comment]
What is strange is that I am the first to detect this problem.
The failure rate is high and most of users should have seen at least that most of channels are not updated.

**Comment by kgoderis** [comment]
I have seen failures on subscriptions because of differences in Sonos hardware (before the change of the thingtypes) (I have play:1, playbar, sub, connect:amp, zp120 - 25 devices in total), and after a restart of the binding but nothing else. Openhab runs on a Mac mini

**Comment by lolodomo** [comment]
Interesting, I discovered that I can have sometimes another error:
```
2016-07-31 13:22:01.128 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Could not find an upnp device for participant 'RINCON_000E58D01E9001400'
2016-07-31 13:22:01.133 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Could not find an upnp device for participant 'RINCON_000E588EAE6601400'
```

**Comment by kgoderis** [comment]
What kind of Sonos device are those 2? Some Sonos do not implement certain UPNP subdevices

**Comment by lolodomo** [comment]
Connect amp and Play 5.

**Comment by lolodomo** [comment]
For the error "Could not find an upnp device for participant", I know what(s wrong. The binding probably invokes actions (or subscribes services) while the jupnp library has not yet discovered the UPnP device.
We have to ckeck if the UPnP device is registered before starting anything in the binding. I will at least change that.

**Comment by lolodomo** [comment]
I have reduced the failure rate by
- moving the GENA subscriptions to the polling job, that is running it asynchronously in a separate thread rather than during initialize() - having too much code inside initialize() could lead to several problems
- delaying GENA subscriptions and actions invokations by the polling job if the UPnP device is not yet registered by jupnp - not frequent but it can happen sometimes

When it happens, it seems that the Sonos is simply not responding to the request. The HttpResponse obkect is null.

The idea now is to implement #1941; then the Sonos binding could be informed if the GENA subscritions succeeded or failed. If it failed, a new subscription will be tried at the next polling cycle.

**Comment by lolodomo** [comment]
Another point I never mentioned: when it fails, it fails for all subscriptions and for all my Sonos (2).
Ok for all or KO for all.

**Comment by kgoderis** [comment]
Interesting. That would point to an issue in the UPNP library

**Comment by lolodomo** [comment]
Here is an exemple of logs:
What is interesting is that the feedback of failures arrive 10 seconds after calling. That would mean that the timeout is around 10 seconds which is more than expected.
Maybe the problem is elsewhere in the jupnp library.
```
21:17:58.714 [DEBUG] [ding.sonos.handler.ZonePlayerHandler] - initializing handler for thing sonos:CONNECTAMP:salon
21:17:58.761 [DEBUG] [ding.sonos.handler.ZonePlayerHandler] - Polling job
21:17:58.762 [DEBUG] [g.sonos.internal.SonosHandlerFactory] - Creating a ZonePlayerHandler for thing 'sonos:PLAY5:chambre' with UDN 'RINCON_000E588EAE6601400'
21:17:58.764 [DEBUG] [ding.sonos.handler.ZonePlayerHandler] - Creating a ZonePlayerHandler for thing 'sonos:PLAY5:chambre'
21:17:58.765 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Registering device '(RemoteDeviceIdentity) UDN: uuid:RINCON_000E58D01E9001400, Descriptor: http://192.168.1.38:1400/xml/device_description.xml' for participant 'RINCON_000E58D01E9001400'
21:17:58.771 [DEBUG] [org.eclipse.smarthome.binding.sonos ] - ServiceEvent REGISTERED - {org.eclipse.smarthome.core.thing.binding.ThingHandler}={thing.type=sonos:PLAY5, thing.id=sonos:PLAY5:chambre, service.id=344, service.bundleid=207, service.scope=singleton} - org.eclipse.smarthome.binding.sonos
21:17:58.807 [DEBUG] [ding.sonos.handler.ZonePlayerHandler] - initializing handler for thing sonos:PLAY5:chambre
21:17:58.812 [DEBUG] [.eclipse.smarthome.io.transport.upnp] - BundleEvent STARTED - org.eclipse.smarthome.io.transport.upnp
21:17:58.825 [DEBUG] [ding.sonos.handler.ZonePlayerHandler] - Polling job
21:17:58.826 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Registering device '(RemoteDeviceIdentity) UDN: uuid:RINCON_000E588EAE6601400, Descriptor: http://192.168.1.13:1400/xml/device_description.xml' for participant 'RINCON_000E588EAE6601400'
21:17:58.875 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Invoking Action 'GetZoneInfo' of service 'DeviceProperties' for participant 'RINCON_000E58D01E9001400'
21:17:58.902 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - Invoking Action 'GetZoneInfo' of service 'DeviceProperties' for participant 'RINCON_000E588EAE6601400'
21:18:09.033 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:ZoneGroupTopology' for device 'uuid:RINCON_000E58D01E9001400' failed: Subscription failed:  No response received. - null - null
21:18:09.035 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:AudioIn' for device 'uuid:RINCON_000E588EAE6601400' failed: Subscription failed:  No response received. - null - null
21:18:09.033 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:GroupManagement' for device 'uuid:RINCON_000E58D01E9001400' failed: Subscription failed:  No response received. - null - null
21:18:09.039 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:GroupManagement' for device 'uuid:RINCON_000E588EAE6601400' failed: Subscription failed:  No response received. - null - null
21:18:09.041 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:DeviceProperties' for device 'uuid:RINCON_000E588EAE6601400' failed: Subscription failed:  No response received. - null - null
21:18:09.041 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:RenderingControl' for device 'uuid:RINCON_000E588EAE6601400' failed: Subscription failed:  No response received. - null - null
21:18:09.042 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:RenderingControl' for device 'uuid:RINCON_000E58D01E9001400' failed: Subscription failed:  No response received. - null - null
21:18:09.044 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:AudioIn' for device 'uuid:RINCON_000E58D01E9001400' failed: Subscription failed:  No response received. - null - null
21:18:09.045 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:ZoneGroupTopology' for device 'uuid:RINCON_000E588EAE6601400' failed: Subscription failed:  No response received. - null - null
21:18:09.045 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:DeviceProperties' for device 'uuid:RINCON_000E58D01E9001400' failed: Subscription failed:  No response received. - null - null
21:18:09.044 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:AVTransport' for device 'uuid:RINCON_000E58D01E9001400' failed: Subscription failed:  No response received. - null - null
21:18:09.046 [DEBUG] [.io.transport.upnp.UpnpIOServiceImpl] - A GENA subscription 'urn:upnp-org:serviceId:AVTransport' for device 'uuid:RINCON_000E588EAE6601400' failed: Subscription failed:  No response received. - null - null
```

**Comment by lolodomo** [comment]
In addition to the GENA subscription problem, I think there is another problem when GENA subscriptions succeeded because for example state and control channels are apparently not always set at startup. Later, when Sonos state is updated, the channels are updated.
It could be either a bug in the binding or a bug in the UPnP transport service / jupnp library that would miss the first GENA notification.
It could even be a bug in the Sonos firmware but I doubt about that because I don't remind of such a problem in the Sonos Vera plugin.

**Comment by lolodomo** [comment]
Ok, understood. The first notifications are bypassed because onValueReceived bypasses everything until the thing is ONLINE. The thing becomes ONLINE later while updatePlayerState is called in the polling job.

@kgoderis : what was the reason to check the thing state in onValueReceived ? In my opinion, if we receive something, we should even make the thing ONLINE if not yet ONLINE.

**Comment by lolodomo** [comment]
Last point is that, to have a correct init state of all channels, the binding should know the group topology first.
Putting service ZoneGroupTopology as first service subscription could help but will not be a garantee that the group topology is loaded first.

**Comment by kgoderis** [comment]
@lolodomo a very long time ago I had a discussion with Kai on how we could check whether devices are online/offline. Unfortunately, I can't remember all the options we discussed, but we decided to implement a separate polling mechanism. But yes, if we receive something we should put the device ONLINE.

**Comment by lolodomo** [comment]
@kgoderis : finally I keep the code unchanged in method onValueReceived but I made a change to be sure that the thing state is initially ONLINE.
You can see the little changes in my PR #1957 waiting for your feedback.

**Comment by kgoderis** [comment]
mmh... that could lead to a strange effect that the device is ONLINE, and then OFFLINE after startup. Maybe it does make more sense to remove https://github.com/lolodomo/smarthome/blob/4beded998c785bb83a52138dd42b2e48a017739f/extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java#L365 altogether. I am not sure what I had that line there in the first place.

**Comment by lolodomo** [comment]
Yes, this test is probably useless.

**Comment by lolodomo** [comment]
Here is the summary of the situation:
- There is probably a bug in the jupnp library making the GENA subscriptions sometimes failing
- With the last PR I submitted (not yet merged), there is a retry mechanism implemented that makes the binding finally reliable
- there is probably still a problem at startup with Sonos things that are slave members of a Sonos group. I have to test this case but I think their channels could sometimes be unset.

**Comment by yphyph01** [comment]
Hi @lolodomo , @kgoderis , @kaikreuzer 
Any update on that subscription issue?
I'm still facing the same error with Sonos ConnactAmp Devices:

`2017-02-05 13:15:07.676 [ERROR] [g.jupnp.protocol.sync.ReceivingEvent] - Invalid subscription ID, no active subscription: (IncomingEventRequestMessage) NOTIFY /upnpcallback/dev/RINCON_B8E937AA001201400/svc/upnp-org/ZoneGroupTopology/event/cb SEQUENCE: 0`

Regards

**Comment by lolodomo** [comment]
You have just to ignore this error produced by the jupnp library. The binding is working well even if you get these errors at startup.

**Comment by yphyph01** [comment]
Thanks for this quick reply.
So, I guess the issues I've got are not link to this one.
If you could see this problem : https://community.openhab.org/t/sonos-playsound-issue-no-sound-and-noerror/20350
Any help / advice would be welcome ;-)

**Comment by yphyph01** [comment]
@lolodomo 

Sorry for the question, but I'd like to know if it's normal to have this infinite loop in the Sonos Binding Debug Trace:

```
2017-02-05 20:07:20.751 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.751 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.762 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.763 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.764 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.765 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.765 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.766 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.767 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.768 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.769 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.769 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.772 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.773 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.774 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.775 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.776 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.776 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.777 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.778 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.779 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.780 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.782 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.783 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.784 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.785 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.786 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.788 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.788 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.788 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.789 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.789 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.789 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.789 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.797 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.798 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.798 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.798 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.799 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.799 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.799 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.799 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.800 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.800 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (Integer) 60 to field 'refresh' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
2017-02-05 20:07:20.802 [DEBUG] [.smarthome.config.core.Configuration] - Setting value (String) RINCON_000E58DA4B7601400 to field 'udn' in configuration class org.eclipse.smarthome.binding.sonos.config.ZonePlayerConfiguration
```

and this is only an extract of 50 milliseconds ;-)

**Comment by lolodomo** [comment]
Excuse me but why pushing that in that issue ?
Discuss first on the forum to identify what you are doing wrong and if an issue is identified, open a new issue describing your issue.

**Comment by yphyph01** [comment]
You're right. My mistake. Sorry
I'll push it on thé forum
Regards

**Comment by kaikreuzer** [comment]
@lolodomo Has this been (at least partially) addressed with #1957?
Is there anything more we can/should do in the binding?

**Comment by lolodomo** [comment]
Yes, there remains potential "wrong informations" at startup in case of a device in a Sonos group. I should have a look, maybe it could be resolved by first requesting the group topology. It is a minor problem that could lead to bad informations in channels.

**Comment by triller-telekom** [comment]
@lolodomo a friendly ping :) Did you have a look at these "wrong information" on start-up? Is it a serious problem or can we close this issue?

**Comment by lolodomo** [comment]
Please keep this issue opened. The bug is not major but not easy to fix and not high in my TODO list.
